extensions/amp-analytics/0.1/config.js

```diff
@@ -64,11 +66,63 @@ export class AnalyticsConfig {
     this.win_ = this.element_.ownerDocument.defaultView;
     this.isSandbox_ = this.element_.hasAttribute('sandbox');
 
-    return this.fetchRemoteConfig_()
+    return Promise.all([this.fetchRemoteConfig_(), this.fetchVendorConfig_()])
       .then(this.processConfigs_.bind(this))
       .then(() => this.config_);
   }
 
+  /**
+   * Constructs the URL where the given vendor config is located
+   * @private
+   * @param {string} vendor the vendor name
+   * @return {string} the URL to request the vendor config file from
+   */
+  getVendorUrl_(vendor) {
+    const isLocalDev = getMode().localDev;
+    const baseUrl = calculateScriptBaseUrl(this.win_.location, isLocalDev);
+    const rtv = isLocalDev ? '' : `/rtv/${getMode().rtvVersion}`;
+    // bg has a special canary config
+    const canary = vendor === 'bg' && isCanary(self) ? '.canary' : '';
```

Comment:
hardcode `bg` looks good to me now. But I do believe we should introduce a more generic check in the future. Any idea? Thanks

Reply:
There isn't anywhere in the code that keeps a list of vendors with canary configs. Currently, everywhere just hard codes the canary configs (maybe 3 different places in the code). Maybe we just need to keep a list of canary vendor configs somewhere `const VENDOR_CANARY_CONFIGS = ['bg']`